### PR TITLE
Gitlab CI migration (9.3-backports)

### PR DESCRIPTION
**Pull request by mliszcz**
This PR adds configuration for Gitlab CI/CD pipelines and introduces necessary test code changes to make them pass on Gitlab.

### CODE REVIEW

**Comment by mliszcz** [comment]
Corresponding MR in gitlab repo: https://gitlab.com/tango-controls/test_gh_import/cppTango/-/merge_requests/820

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    OS_TYPE: debian8
+    # SONAR_SCANNER: "ON"
+    # COVERALLS: "ON"
+    STOCK_CPPZMQ: "OFF"
+
+debian7:
+  <<: *job-template
+  variables:
+    <<: *variables
+    OS_TYPE: debian9
```
Comment (t-b) [inline .gitlab-ci.yml]: Typo? debian9 -> debian7

Comment (mliszcz) [reply .gitlab-ci.yml]: Fixed, thanks for noticing. Let's see if it builds: https://gitlab.com/tango-controls/test_gh_import/cppTango/-/jobs/966103473

**Review by t-b — DISMISSED** [review]

**Comment by mliszcz** [comment]
Hi @t-b, I needed to rebase the PR on target branch due to the sed fix being merged. You need to re-check once again. Sorry for the inconvenience!

**Review by t-b — APPROVED** [review]

**Pull request merged**